[Debbie Wu] iP

debwy opened the pull request:
### Dave
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Your mind will be freed by Dave. Dave will...
- Communicate via text
- Be easy to pick up
- Be **mediocre** in terms of speed

If you feel like it, you can
1. _Download_ it from [here](https://se-education.org/guides/conventions/java/intermediate.html)
2. _That's all I think_

**FREE!**

Features:
- [x] Task management
- [ ] Deadline management
- [ ] Reminders

Here is a Java `main` method
`
public class Main {
    public static void main(String[] args) {
        // Hello World
    }
}
`




```Java
public class Main {
    public static void main(String[] args) {
        // Hello World
    }
}
```

Commit:
Add Gradle support

debwy committed:
Level-1

debwy committed:
Level-2

debwy committed:
Level-3

debwy committed:
Level-4

debwy committed:
A-TextUiTesting

debwy committed:
Level-5

debwy committed:
Level-6

debwy committed:
Small changes

debwy committed:
Add save/load functionality (Level-7)

debwy committed:
Add date and times for deadline

debwy committed:
Merge pull request #1 from debwy/branch-Level-7

Add save/load functionality (Level-7)

debwy committed:
Merge pull request #2 from debwy/branch-level-8

Add date and times for deadline

debwy committed:
Fix bug from merging

debwy committed:
Separate main class into smaller classes

debwy committed:
Organize into packages

debwy committed:
Merge pull request #3 from debwy/add-gradle-support

Add Gradle support

debwy committed:
Add gradle

debwy committed:
Add JUnit tests

therealdaofu commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,177 @@
+        }
+    }
+
+    private void execute(Command c, String s) throws IOException, DukeTaskException {
```
I really like this function, looks clean and easy to understand

therealdaofu commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,177 @@
+                        }
+                        int index = Integer.parseInt(restWord) - 1; //array starts from 0
+                        Task temp = log.get(index);
+                        temp.Unmark();
```
It looks like you are not using this chunk of code anymore, maybe you could consider deleting it? or just copy it to a different file just in case

therealdaofu commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,66 @@
+
+        int indexOfSpace = s.indexOf(' ');
+        isMultipleWords = indexOfSpace > -1;
+        firstWord = s;
```
Perhaps you can initialise both firstword and restWord at lines 9 & 10

therealdaofu commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,84 @@
+        }
+    }
+
+    private void parse(char c, String s) throws DukeException {
```
Might be cleaner if you use switch case here

therealdaofu commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,80 @@
+        return log;
+    }
+
+    public Task addTodo(String s) throws DukeTaskException {
```
Is it possible to generalise addTodo etc. to maybe something like addTask(Task newTask)?

zlimez commented on src/main/java/main.java:
```diff
@@ -0,0 +1,7 @@
+import duke.Duke;
+
+public class main {
```
```suggestion
public class Main {
```
I really like that you extracted the entry point of the application into its own class. It is very clean. Hmm but perhaps the class name main could be capitalised to follow the coding standard?

zlimez reviewed (commented):
Small coding standard review otherwise fantastic

debwy committed:
Create a JAR File

debwy committed:
Add JavaDoc comments

debwy committed:
Tweak code to comply with the coding standard & added suggestions from the pr review

debwy committed:
Add feature: Find tasks using keyword(s)

debwy committed:
Merge pull request #5 from debwy/branch-Level-9

Add feature: Find tasks using keyword(s)

debwy committed:
Merge pull request #6 from debwy/branch-A-CodingStandard

Follow coding standard & added suggestions

debwy committed:
Merge branch 'master' into branch-A-JavaDoc

debwy committed:
Merge pull request #7 from debwy/branch-A-JavaDoc

Add JavaDoc comments

debwy committed:
Add a GUI to Duke with JavaFX

debwy committed:
Edit build.gradle

debwy committed:
Minor changes

debwy committed:
Merge pull request #8 from debwy/branch-level-10

Minor changes

debwy committed:
Add assertions

Assertions were used in files Duke, TaskList & Task to ensure that
variables are within their expected values.

Clean up some pieces of code such as,
* Delete unused methods/code from CLI implementation in Duke.java
* Update Javadoc in Tasklist.java

debwy committed:
Improve code quality

Check that code adheres to the required code quality to make it
more reader-friendly.

Minor improvements
* Made deadline constructor error more descriptive for the user
* Deleted more unused code
* Rename test method names
* Added missing/more Javadoc

debwy committed:
Merge pull request #9 from debwy/branch-A-Assertions

Add assertions

debwy committed:
Merge branch 'master' into branch-A-CodeQuality

debwy committed:
Merge pull request #10 from debwy/branch-A-CodeQuality

Improve code quality

debwy committed:
Add Fixed Duration Tasks (B) and Better Search (C)

debwy committed:
Add screenshot of User Interface

debwy committed:
Add a ReadMe

debwy committed:
Make small changes
